### ISSUE TITLE
Bruteforce attack dosen't work (Angular 13)

### ISSUE THREAD

**Issue by Webkinger** [cryptic-game/frontend#345]
**Describe the bug**
The bruteforce attack cannot be performed

**Reproduce**

![Screenshot 2022-03-13 181053](https://user-images.githubusercontent.com/33805308/158070951-746417ce-bc74-400c-918a-c95aa2f425b1.png)

![Screenshot 2022-03-13 213908](https://user-images.githubusercontent.com/33805308/158078342-6244ea9d-cb20-484f-8a96-5957a0c4d255.png)

**Comment by use-to** [comment]
There is also a bigger problem here. The bruteforce command triggers a request to `/bruteforce/status` (before actually trying to initialize the attack via `/bruteforce/attack`) which returns in your case something like this:
```json
{
  "data": {
    "error": "attack_not_running"
  }, 
  "tag": "..."
}
```
The frontend tries to catch errors directly and pass them into the subscription error.
But the ms-service is just expecting errors stored on a higher level inside the payload.
Such as:
```json
{
  "data": {"...": "..."},
  "error": "attack_not_running",
  "tag": "..."
}
```
Its totally inconsistent and just not well defined - which leads to a bigger problem at this point.
Also interesting is the fact that `terminal-state.ts` consists includes 1700 loc.

**Comment by use-to** [comment]
Because no error was passed to the subscription, the status subscription triggers the normal behavior which tries to print out the `target_device` using `promptAppender` [(see line 1159)](https://github.com/cryptic-game/frontend/blob/master/src/app/desktop/windows/terminal/terminal-states.ts#L1159).
The real bug occurs inside the `handleMessage` function in `websocket.service.ts` [(starting at line 184)](https://github.com/cryptic-game/frontend/blob/master/src/app/websocket.service.ts#L184) or inside the microservice implementing this endpoint - not sure because of the missing provision of documentation here.

**Comment by use-to** [comment]
Also some players seem to not having the problem. I think its just a (in this case positive) side-effect of the very strong caching-strategy (sw and maybe http-middleware) used in production.

**Comment by Tristan-H11** [comment]
Referring to the wiki (German: https://wiki.cryptic-game.net/books/einf%C3%BChrungen/page/websocket-api-endpunkte), the frontend does everything correctly when trying to catch errors on the first level.
An answer should be structured as followed:
```json
{
 "status": {
   "code": 200,
   "name": "OK"
 },
 "error": "<message>", // optional
 "tag": "some_tag",
 "data": {
   "id": "fd973bf5-3801-402f-8c85-d5313aa02baa",
   "user_id": "2cea82a0-3425-4d0e-8e1b-4823ad744162"
 }
}
```
Thus, there must be an error in the creation of the WS-Answer. @cryptic-game/java 

As use-to already mentioned, please provide a broader set of endpoint examples and more documentation. @cryptic-game/backend

**Comment by MarcelCoding** [comment]
@Tristan-H11 thats the old v3 documentation.
But the v2 documentation says the same about errors: https://github.com/cryptic-game/server/wiki/API-Reference#contact-microservice
https://github.com/cryptic-game/cryptic-service/wiki/API-Reference-%5Bexperimental%5D#bruteforce-service

**Comment by Tristan-H11** [comment]
Oh. Did not know where to find this documentation. Is there a place, where all these different wikis are linked? @MarcelCoding 

If I'm understanding this correctly, that's a backend error, right?

**Comment by McModknower** [comment]
Because of the check in https://github.com/cryptic-game/frontend/blob/1344edf8040c435ecf4de25137b0f5b58b70d558/src/app/websocket.service.ts#L188 just putting the error message on the top level won't work.

In general that line and https://github.com/cryptic-game/frontend/blob/1344edf8040c435ecf4de25137b0f5b58b70d558/src/app/websocket.service.ts#L197 both check for `message['error']`, and if error is not null, it will never reach the second one.

That means no matter what the backend does, the bruteforce attack will only work if something in the frontend changes.

Maybe the check in line 188 shouldn't be there. But I don't know where the documentation about how it should work is.

**Comment by MarcelCoding** [comment]
fixed by #355
